### PR TITLE
finished spBestSIFTL2SquaredDistance

### PR DESCRIPTION
**Pull request by bar-sc**
Hi, just finished the function spBestSIFTL2SquaredDistance.

חג שמח!

### CODE REVIEW

File: sp_image_proc_util.cpp [hunk 1]
```diff
@@ -40,38 +67,85 @@ double **spGetSiftDescriptors(char *str, int maxNFeautres, int *nFeatures) {
     *nFeatures = (int) kp1.size();
 
     //Allocate memory for 2d array
-    // ***NEED TO FREE SOMEWHERE!!!***
-    // ***NEED TO FREE SOMEWHERE!!!***
-    // ***NEED TO FREE SOMEWHERE!!!***
     double **matArr = (double **) malloc(sizeof(double *) * ds1.rows);
-    if (matArr == NULL) { return NULL; };
+    if (matArr == NULL) { printf("An error occurred - allocation failure\n"); };
```
Comment (RedBird) [inline sp_image_proc_util.cpp]: Where did they say what to do on this cases?

File: main.cpp [hunk 2]
```diff
@@ -4,8 +4,37 @@
...
-    double **mat = spGetSiftDescriptors(src, 1000, &nFeatures);
-}
\ No newline at end of file
+    int nFeatures1=0;
+    int nFeatures2=0;
+
+    char src1[] = "/home/bar/ClionProjects/Software-Project-ex2/images/img2.png";
+    char src2[] = "/home/bar/ClionProjects/Software-Project-ex2/images/img8.png";
+
+    //problem with img1.png (nFeatures1 show 451 instead of 450)
```
Comment (RedBird) [inline main.cpp]: What does this problem mean?

File: sp_image_proc_util.cpp (hunk 1 above)
Comment (bar-sc) [inline sp_image_proc_util.cpp]: In the header file there are some more instructions

File: main.cpp (hunk 2 above)
Comment (bar-sc) [inline main.cpp]: There is a very mysterious bug. I'll open an issue here later

**Pull request merged**